lib/llvmopencl/TargetAddressSpaces.cc

```diff
@@ -565,6 +565,11 @@ TargetAddressSpaces::runOnModule(llvm::Module &M) {
     addrSpaceMapDown[POCL_AS_FAKE_GLOBAL] = 1;
     addrSpaceMapDown[POCL_AS_FAKE_LOCAL] = 3;
     addrSpaceMapDown[POCL_AS_FAKE_CONSTANT] = 2;
+  } else if (arch.startswith("nvptx")) {
```

Comment:
You might want to try if NVPTX works without the fake address spaces now. By default we don't use them anymore and it's likely that the code path will be phased out. TAS pass is specifically fragile and the main reason I wanted to get rid of the fake ids.